src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -117,6 +117,13 @@ private static IEnumerable<Measurement<long>> GetGarbageCollectionCounts()
 #if NET6_0_OR_GREATER
         private static IEnumerable<Measurement<long>> GetFragmentationSizes()
         {
+            if (GC.CollectionCount(0) == 0)
+            {
+                // gen0 collection has never happened before,
+                // do not report the measurements as GC doesn't have this information yet
+                return new Measurement<long>[] { };
```

Comment:
Not sure if it worth to extract this to a constant (to save the alloc). Probably no because it's only triggered at low frequency (during exporter).

Reply:
👮 I'm issuing you a citation for violation of [CA1825: Avoid zero-length array allocations](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1825).

```csharp
return Array.Empty<Measurement<long>>();
```

😄

Reply:
Thanks! This is much better. I'll see if we can add SA1825 in the CI.